### ISSUE TITLE
Questions about the update of policy model

### ISSUE THREAD

**Issue by ZetangForward** [yifan123/flow_grpo#28]
Hi, thanks for your great work. I have a question about the updating rules of policy model.
As shown in your code: https://github.com/yifan123/flow_grpo/blob/ed4e82b5c89a0830cf40d7f3f97892a7f79f004f/scripts/train_sd3.py#L967

This loss backward is wrapped in the for loop, indicating that the policy model is updated in every `for` loop.

https://github.com/yifan123/flow_grpo/blob/ed4e82b5c89a0830cf40d7f3f97892a7f79f004f/scripts/train_sd3.py#L913-L919

And this `for` loop represents a step-by-step process of denosing for a batch of samples, does this mean that for each sample, each denoising step needs to be optimized once?

Is this a bit inefficient for training?

Also I've noticed that other issues have looked at training efficiency as well: https://github.com/yifan123/flow_grpo/issues/22

Looking forward for your reply, thx @yifan123

**Comment by yifan123** [comment]
The actual gradient accumulation is performed with `gradient_accumulation_steps = config.train.gradient_accumulation_steps * num_train_timesteps` [code](https://github.com/yifan123/flow_grpo/blob/main/scripts/train_sd3.py#L385 ). For each sample, it needs to be optimized once.

**Comment by ZetangForward** [comment]
> As indicated in the [issue](https://github.com/yifan123/flow_grpo/issues/22#issuecomment-2933111666), for a 2-node configuration, each epoch takes approximately 0.05 hours to complete on A800. Therefore, a flux model update time of around 3 minutes seems reasonable.

Hi, thx for your reply. 

I observed that there are 50,000 samples inside https://github.com/yifan123/flow_grpo/blob/main/dataset/geneval/train_metadata.jsonl, but according to the provided training script, it is not possible to go through these samples in 2 days' time. Can you please tell me how many samples are trained to get a significant improvement in results?
